Whitelist commit-lockfile-summary in flake nixConfig (#8220)
NixOS/nix: src/nix/flake.md

* `nixConfig`: a set of `nix.conf` options to be set when evaluating any
  part of a flake. In the interests of security, only a small set of

>   whitelisted options (currently `bash-prompt`, `bash-prompt-prefix`,
>   `bash-prompt-suffix`, `flake-registry`, and `commit-lockfile-summary`)
>   are allowed to be set without confirmation so long as `accept-flake-config`
>   is not set in the global configuration.

Reviewer: Given this is a substantial list now, it makes sense to present it as such. It also reveals that the sentence is a bit clumsy and should be reworded:

```suggestion
set of options is allowed to be set without confirmation so long as [`accept-flake-config`](@docroot@/command-ref/conf-file.md#conf-accept-flake-config) is not enabled in the global configuration:
   - [`bash-prompt`](@docroot@/command-ref/conf-file.md##conf-bash-prompt)
   - [`bash-prompt-prefix`](@docroot@/command-ref/conf-file.md#conf-bash-prompt-prefix)
   - [`bash-prompt-suffix`](@docroot@/command-ref/conf-file.md#conf-bash-prompt-suffix)
   - [`flake-registry`](@docroot@/command-ref/conf-file.md#conf-flake-registry)
   - [`commit-lockfile-summary`](@docroot@/command-ref/conf-file.md#commit-lockfile-summary)
```

This should make those options much more discoverable.

Reviewer 2: Can't commit in GitHub UI: "Applying suggestions on deleted lines is currently not supported."
Fixed two types and committed by hand.
